### ISSUE TITLE
Items with guide flag

### ISSUE THREAD

**Issue by philkingston** [paperjs/paper.js#195]
I'm trying to use the 'guide' flag on various items in my project to prevent them from being targetting by a HitResult.

I have a basic Item (paraphrasing):

```
line = new Path.Line(...);
line.guide = true;
```

I don't want it caught by any hitTests, so I set guide : false in the hitOptions. Here is how I call the hitTest:

```
var hitResult = project.hitTest(event.point, {
    ...
    guide : false,
    ...
});
```

This still brings back Items that I have specifically flagged as guides. The only way I have managed to exclude them from the selection is by adding the following to paper.js:

```
if(!options.guide && this._guide)
        return null;
```

Here's the code in situ:

```
_hitTest: function(point, options) {
    var style = this._style,
        tolerance = options.tolerance || 0,
        radius = (options.stroke && style._strokeColor
                ? style._strokeWidth / 2 : 0) + tolerance,
        loc,
        res;
    var coords = [],
        that = this;
    if(!options.guide && this._guide)
        return null;
    function checkPoint(seg, pt, name) {
        if (point.getDistance(pt) < tolerance)
            return new HitResult(name, that, { segment: seg, point: pt });
    }
    function checkSegment(seg, ends) {
        var point = seg._point;
        return (ends || options.segments)
                && checkPoint(seg, point, 'segment')
            || (!ends && options.handles) && (
                checkPoint(seg, point.add(seg._handleIn), 'handle-in') ||
                checkPoint(seg, point.add(seg._handleOut), 'handle-out'));
    }
    if (options.ends && !options.segments && !this._closed) {
        if (res = checkSegment(this.getFirstSegment(), true)
                || checkSegment(this.getLastSegment(), true))
            return res;
    } else if (options.segments || options.handles) {
        for (var i = 0, l = this._segments.length; i < l; i++) {
            if (res = checkSegment(this._segments[i]))
                return res;
        }
    }
    if (options.stroke && radius > 0)
        loc = this.getNearestLocation(point);
    if (!(loc && loc._distance <= radius) && options.fill
            && this.hasFill() && this.contains(point))
        return new HitResult('fill', this);
    if (!loc && options.stroke && radius > 0)
        loc = this.getNearestLocation(point);
    if (loc && loc._distance <= radius)
        return options.stroke
                ? new HitResult('stroke', this, { location: loc })
                : new HitResult('fill', this);
}
```

I then get the exact behaviour I expect.

First of all, is this the right approach for 'guide' objects? Is anybody else using the guide flag successfully without this patch? If the patch is required, it may well need to be added to all polymorphed _hitTest methods, as I've currently only changed the Path class.

Thanks,
Phil

**Comment by NerdGr8** [comment]
This is still an issue. I just used the option :
`this.hitItem = paper.project.hitTest(event.point, { fill: true, stroke: true, pixel: true, tolerance: hitSize, guides: false });`

And it still hit tests guides

**Comment by eeropic** [comment]
Cannot reproduce this, 
[example sketch with item.guide:true and hittest with guides:false](http://sketch.paperjs.org/#V/0.11.8/S/XZDNasMwDMdfxfiUgQkZ7OTRU19gsN3qHjxHbbS4krEVWgh999ltx9gEAumvj5/QqsmfQFv9PoOESRsdeGx5BFHJy7QhOKu3GvRbzCFCtzpS1RIXFGSyu+dhMNX35l7IfsSl2JfhkR8wxi1HztbpDKPTD/244AhW8gKOrk+OHDXkhLJJmb8gSF/jDyjStTP6H55R6+9eq9q8UXelSOYZ/moJL/Cv7QYuVh18LKAa/LXRA1PhCH3kY1fJt4vqOz4z+DkxkhRtd/vrNw==)

**Comment by eeropic** [comment]
Verified - the last version where this occurs is 0.9.25
http://sketch.paperjs.org/#V/0.9.25/S/bVDLasMwEPwVoZMNwrjQk0pO6bUQaG9RDqq8jlWru0KPpBD875Vilx5aCcHs7KCZ3RtH/Qlc8tcZkpm44IaGWl90YF6naYdwZYcCur0NxkFzU8jK8RRtsoTy+ND3oryTWBtBDzZH+dhv9Wid25OjIBUPMCi+8edsB5ApZFC4tAoVjhlN/ZIRvlCO8ExXbOACmNrNtIaabNr5QB9gUlfwG8S0ijpPFpNgm/bHW7LqIdgvG1OgGf7y3n7BP/J70CjZqF2ElV7apxUYwkgOOkfnpoQpYyx1knrLKt8D6PmeKnJ5PC3f

Closing the issue as this has been fixed in 0.10.2
